From da28d7eace3c0c119479eb69d8c7373664640a25 Mon Sep 17 00:00:00 2001
From: Rechi <Rechi@users.noreply.github.com>
Date: Sat, 17 Aug 2019 20:29:02 +0200
Subject: [PATCH] [fix] python crashes after bf0834e

Incrementing reference counting was missing before, so global Py_True & Py_False got freed.
---
 xbmc/interfaces/python/PythonSwig.cpp.template | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/interfaces/python/PythonSwig.cpp.template b/xbmc/interfaces/python/PythonSwig.cpp.template
index 1f8d44aa437e..0a69b5aa3c7b 100644
--- a/xbmc/interfaces/python/PythonSwig.cpp.template
+++ b/xbmc/interfaces/python/PythonSwig.cpp.template
@@ -47,7 +47,7 @@ Helper.setup(this,classes,
     [ 'void' : 'Py_INCREF(Py_None);\n    ${result} = Py_None;',
       'long': '${result} = PyInt_FromLong(${api});',
       'unsigned long': '${result} = PyInt_FromLong(${api});',
-      'bool': '${result} = ${api} ? Py_True : Py_False;',
+      'bool': '${result} = ${api} ? Py_True : Py_False; Py_INCREF(${result});',
       'long long': '${result} = Py_BuildValue("L", ${api});',
       'int': '${result} = Py_BuildValue("i", ${api});',
       'unsigned int': '${result} = Py_BuildValue("I", ${api});',
